src/main/kotlin/org/opensearch/replication/action/stats/FollowerStatsResponse.kt

```diff
@@ -0,0 +1,151 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.replication.action.stats
+
+
+import org.apache.logging.log4j.LogManager
+import org.opensearch.action.FailedNodeException
+import org.opensearch.action.support.nodes.BaseNodesResponse
+import org.opensearch.cluster.ClusterName
+import org.opensearch.common.Strings
+import org.opensearch.common.io.stream.StreamInput
+import org.opensearch.common.io.stream.StreamOutput
+import org.opensearch.common.xcontent.ToXContent.EMPTY_PARAMS
+import org.opensearch.common.xcontent.ToXContent.Params
+import org.opensearch.common.xcontent.ToXContentFragment
+import org.opensearch.common.xcontent.XContentBuilder
+import org.opensearch.common.xcontent.XContentFactory
+import org.opensearch.index.shard.ShardId
+import org.opensearch.replication.metadata.ReplicationOverallState
+import org.opensearch.replication.metadata.state.REPLICATION_LAST_KNOWN_OVERALL_STATE
+import org.opensearch.replication.metadata.state.ReplicationStateMetadata
+import org.opensearch.replication.task.shard.FollowerIndexStats
+import org.opensearch.replication.task.shard.FollowerShardMetric.FollowerShardStats
+import java.io.IOException
+
+class FollowerStatsResponse : BaseNodesResponse<FollowerNodeStatsResponse?>, ToXContentFragment {
+    var shardStats :MutableMap<ShardId, FollowerShardStats> = mutableMapOf()
+    var indexStats :MutableMap<String, FollowerIndexStats> = mutableMapOf()
+
+    var pausedIndices :Int = 0
+    var failedIndices :Int = 0
+    var bootstrappingIndices :Int = 0
+    var syncingIndices :Int = 0
+    var shardTaskCount :Int = 0
+    var indexTaskCount :Int = 0
+
+    var opsWritten: Long = 0
+    var opsWriteFailures: Long = 0
+    var opsWriteThrottles: Long = 0
+    var opsRead: Long = 0
+    var opsReadFailures: Long = 0
+    var opsReadThrottles: Long = 0
+    var localCheckpoint: Long = 0
+    var remoteCheckpoint: Long = 0
+    var totalWriteTime: Long = 0
+
+    companion object {
+        private val log = LogManager.getLogger(FollowerStatsResponse::class.java)
+    }
+
+    constructor(inp: StreamInput) : super(inp) {
+        shardStats = inp.readMap(::ShardId, ::FollowerShardStats)
+    }
```

Comment:
Is it possible to always derive from shardStats and only store that?

Main concerns on storing the same data twice is potential paths to introduce bugs where the paths are inconsistently handled.

Reply:
Added the same in latest revision.